streams/src/main/java/org/apache/kafka/streams/kstream/internals/KGroupedStreamImpl.java

```diff
@@ -117,14 +117,26 @@
 
     @Override
     public KTable<K, Long> count() {
-        return count(Materialized.<K, Long, KeyValueStore<Bytes, byte[]>>with(keySerde, Serdes.Long()));
+        return doCount(Materialized.with(keySerde, Serdes.Long()));
     }
 
     @Override
     public KTable<K, Long> count(final Materialized<K, Long, KeyValueStore<Bytes, byte[]>> materialized) {
         Objects.requireNonNull(materialized, "materialized can't be null");
+
+        // TODO: remove this when we do a topology-incompatible release
+        // we used to burn a topology name here, so we have to keep doing it for compatibility
+        final String givenStoreName = new MaterializedPeek<>(materialized).givenStoreName();
```

Comment:
This logic is repeated in a couple of places.
I'm wondering if we could change `MaterializedPeek` to take the `InternalSteamsBuilder` as an additional constructor param and have the logic inside the class, and this block of code could be replaced with  `new MaterializedPeek<>(materialized, builder).maybeIncrementTopologyCount()` or something like that.

Reply:
I get where you're coming from, but I think the duplication is justified in this case. Aside from the three examples in this diff, we interact with Materialized and MaterializedInteral in dozens of places. What we're doing here is very specific to a regression in these three methods.

I think that if we were to separate the regression patch from these locations, we risk having it misused elsewhere. If we did consolidate it, I would want to name it something like `new CountAggregationRegression4919Patch(materialized, builder).maybeIncrementTopologyCount()` to be sure it's used properly.

My experience is that deduplicating code often results in a net increase in complexity and is therefore not without cost. I'm not saying we should have copy-pasted code all over the place, just that we shouldn't universally apply a deduplication strategy. It seems like this is an example where it's better to just have the logic applied where it's needed.